docs/developer-docs/latest/development/backend-customization/controllers.md

```diff
@@ -277,6 +273,96 @@ async delete(ctx) {
 ::::
 :::::
 
+::::: details Collection type examples (Replacing a core action)
+
+:::: tabs card
+
+::: tab find()
+
+```js
+async find(ctx) {
+  const { query } = ctx;
+  // some logic here
+  const entity = await strapi.service('api::restaurant.restaurant').find(query);
+  const sanitizedEntity = await this.sanitizeOutput(entity, ctx);
+  // some logic here
+
+  return this.transformResponse(sanitizedEntity.results, {
+    pagination: sanitizedEntity.pagination,
+  });
+}
+```
+
+:::
+
+::: tab findOne()
+
+```js
+async findOne(ctx) {
+  const { id } = ctx.params;
+  const { query } = ctx;
+  // some logic here
+  const entity = await strapi.service('api::restaurant.restaurant').findOne(id, query);
+  const sanitizedEntity = await this.sanitizeOutput(entity, ctx);
+  // some logic here
+
+  return this.transformResponse(sanitizedEntity);
+}
+```
+
+:::
+
+::: tab create()
+
+```js
+async create(ctx) {
+  const { request } = ctx;
+  // some logic here
+  const entity = await strapi.service('api::restaurant.restaurant').create(request.body);
+  const sanitizedEntity = await this.sanitizeOutput(entity, ctx);
+  // some logic here
+
+  return this.transformResponse(sanitizedEntity);
+}
+```
+
+:::
+
+::: tab update()
```

Comment:
Same here, doesn't take into account if files are passed nor will this work to have specific query params to be returned with the updated information (population, field selection, ect)

see: 

https://github.com/strapi/strapi/blob/09757686b0ddaec8609e2a5b657e6159a0a8fe5e/packages/core/strapi/lib/core-api/controller/collection-type.js#L69-L92